pkg/blockbuilder/builder/builder.go

```diff
@@ -592,7 +616,7 @@ func (i *BlockBuilder) loadRecords(ctx context.Context, c *kgo.Client, partition
 		}
 	}
 
-	return lastConsumedOffset, boff.Err()
+	return lastSeenOffset, boff.Err()
```

Comment:
nit: this could return `offset.Max` if it ends up in this [branch](https://github.com/grafana/loki/blob/4a8fde014cad6efff4449c7fb3c78763864e39a1/pkg/blockbuilder/builder/builder.go#L584). but i think this method should return the last consumed offset instead.

should we instead set `lastSeenOffset` [here](https://github.com/grafana/loki/blob/4a8fde014cad6efff4449c7fb3c78763864e39a1/pkg/blockbuilder/builder/builder.go#L593) only after processing the record

Reply:
Yeah that's a fair point. The reason I removed this was because outside of logging, it's not _used_ anywhere. It's returned up the chain and logged, but it doesn't influence behavior. I figured reducing unnecessary complexity made sense in this case. The new behavior is we're logging the last _seen_ offset, rather than the last _consumed_ offset. However, I should probably just log the final consumed offset (rather than seen), but not return thread it through our code unnecessarily.